src/config.py

```diff
@@ -93,6 +93,9 @@ class Config:
 
     PQL_CUSTOM_METHODS = parse_and_import_custom_methods(PLUGINS_CONFIG_PATH)
 
+    EVM_CHAIN_REFERENCE_DATA = json.load(open("src/data/evm_chains.json"))
+    VALIDATE_CHAINS = getenv("VALIDATE_CHAINS", "True") == "True"
```

Comment:
I introduced the `VALIDATE_CHAINS` env var purely for testing purposes so yes it would have made sense to use `TESTING`, I just didn't notice it.  

If we move the chain validation from the chain constructor back to the `get_connection` method then it will not cause an issue with testing as the get_connection method is not called in any of our tests.  We could then potentially use `VALIDATE_CHAINS` env var to populate the `validate_chain` argument when calling the `get_connection(self, validate_chain: bool)` method (if we decide to add the `validate_chain` argument to the method).

After looking over the celery tasks I now realise why we would want to avoid calling the `validate_chain` method in the constructor.   Moving this responsibility to `get_connection` makes sense for now.  As you mentioned previously this will be more appropriately placed in the object concerned with managing the global connection pool once it is created.

Reply:
Hope you feel better soon!!